webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,399 @@
-let m = require('mithril');
-let rs = require('rswebui');
-let widget = require('widgets');
-let people_util = require('people/people_util')
+let m = require("mithril");
+let rs = require("rswebui");
+let widget = require("widgets");
+let people_util = require("people/people_util");
 
+function checksudo(data) {
+  if (data === "0000000000000000") return true;
+  return false;
+}
+
+const SignedIdentiy = () => {
+  let passphase = "";
+  let id = "";
+  return {
+    view: (v) => [
+      m("i.fas.fa-user-edit"),
+      m("h3", "Enter your passpharse"),
+      m("hr"),
+
+      m("input[type=password][placeholder=Passpharse]", {
+        style: "margin-top:50px;width:80%",
+        oninput: (e) => {
+          passphase = e.target.value;
+        },
+      }),
+      m(
+        "button",
+        {
+          style: "margin-top:160px;",
+          onclick: () => {
+            new Promise(function (resolve, reject) {
+              rs.rsJsonApiRequest("/rsIdentity/getOwnSignedIds", {}, (owns) => {
+                owns.ids.length > 0 ? resolve(owns.ids[0]) : resolve(id);
+              });
+            })
```

Comment:
No, it already returns a promise, wrapping it is not required. You can simply do:
```js
rs.rsJsonApiRequest(...).then(...
```
It uses [m.request](https://mithril.js.org/request.html) under the hood.

You don't have to pass a callback here either, the `ids.length` check can be done inside `then`.

Reply:
@rottencandy is it ok now 

``` javascript 
 rs.rsJsonApiRequest("/rsIdentity/getOwnSignedIds", {}, (owns) => {
                owns.ids.length > 0
                  ? rs.rsJsonApiRequest(
                      "/rsIdentity/createIdentity",
                      {
                        id: owns.ids[0],
                        name: v.attrs.name,
                        pseudonimous: false,
                        pgpPassword: passphase,
                      },
                      (data) => {
                        const message = data.retval
                          ? "Successfully created identity."
                          : "An error occured while creating identity.";
                        widget.popupMessage([
                          m("h3", "Create new Identity"),
                          m("hr"),
                          message,
                        ]);
                      }
                    )
                  : widget.popupMessage([
                      m("h3", "Create new Identity"),
                      m("hr"),
                      "An error occured while creating identity.",
                    ]);
              });
```

Reply:
Yes, this callback method is also fine.

Reply:
Thanks, @rottencandy. I  added these changes in this  [commit](https://github.com/RetroShare/RSNewWebUI/pull/23/commits/6fbf24ee3f601ff723f7903da1567faf83c62609).  Once Please check it out and approve this pull request.